### PR TITLE
Convert docs to Markdown

### PR DESCRIPTION
**Pull request by blowmage**
Convert many elements in the documentation from RDoc to Markdown

- [x] headings
- [x] links
- [x] formatted text
- [x] code blocks

### CODE REVIEW

**Comment by blowmage** [comment]
This is good to go. You can see the how the docs are being build on my personal repo here: http://blowmage.com/gcloud-ruby/docs/master/index.html

File: lib/gcloud/bigquery/load_job.rb [hunk 1]
```diff
@@ -56,7 +56,7 @@ def delimiter
 
       ##
       # The number of header rows at the top of a CSV file to skip. The default
-      # value is +0+.
+      # value is \`0\`.
```
Comment (quartzmo) [inline lib/gcloud/bigquery/load_job.rb]: Accidental backslashes?

Comment (blowmage) [inline lib/gcloud/bigquery/load_job.rb]: Yep. Good catch. Fixed.

**Comment by quartzmo** [comment]
Yay for markdown! Thanks @blowmage!

**Pull request merged**